modules/core/src/main/java/org/apache/ignite/internal/processors/cache/tree/updatelog/UpdateLogRow.java

```diff
@@ -113,7 +113,7 @@ UpdateLogRow initRow(CacheGroupContext grp) throws IgniteCheckedException {
 
     /** Updated entry key. */
     public KeyCacheObject key() {
-        return rowData.key();
+        return rowData == null ? null : rowData.key();
```

Comment:
Let's revert this.
If there is a place where NPE happens, then we shouldn't call this method.